### DISCUSSION TITLE
How to get option's description?

### DISCUSSION THREAD

**Discussion by Freed-Wu** [mesonbuild/meson#13139] [Q&A]
For example:

`meson.options`:

```meson
option(
  'dry_run',
  type: 'boolean',
  value: false,
  description: 'do not write any file',
)
```

`meson.build`:

```meson
conf_data = configuration_data()
conf_data.set('DRY_RUN', get_option('dry_run'), description: get_option_description('dry_run'))
```

Or:

```meson
conf_data = configuration_data()
conf_data.set('DRY_RUN', option: get_option_value_and_description('dry_run'))
```

**Comment by dcbaker** [comment]
You can't, those descriptions are currently only used by Meson when you run `meson configure` to print descriptions to the console. Patches welcome to add support for getting the description. I'd have a preference for a keyword to `get_option` over a new function
